File: src/main/java/net/gegy1000/terrarium/server/map/source/raster/ShortRasterSource.java
package net.gegy1000.terrarium.server.map.source.raster;

import net.gegy1000.terrarium.server.util.Coordinate;

public interface ShortRasterSource {
    short get(Coordinate coordinate);

    void sampleArea(short[] data, Coordinate minimumCoordinate, Coordinate maximumCoordinate);
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/source/raster/ShortRasterSource.kt
package net.gegy1000.terrarium.server.map.source.raster

import net.gegy1000.terrarium.server.util.Coordinate

interface ShortRasterSource {
    fun get(coordinate: Coordinate): Short

    fun sampleArea(result: ShortArray, minimumCoordinate: Coordinate, maximumCoordinate: Coordinate)
}
